pkg/cloudmap/client.go

```diff
@@ -208,6 +201,17 @@ func (sdc *serviceDiscoveryClient) getEndpoints(ctx context.Context, nsName stri
 		return endpts, nil
 	}
 
+	endpts, err = sdc.getEndpointsInThisCluster(ctx, nsName, svcName, false)
+	if err != nil {
+		sdc.log.Error(err, "failed to retrieve clusterSetId")
+		return nil, err
+	}
+	sdc.cache.CacheEndpoints(nsName, svcName, endpts)
+
+	return endpts, nil
+}
+
+func (sdc *serviceDiscoveryClient) getEndpointsInThisCluster(ctx context.Context, nsName string, svcName string, inThisCluster bool) (endpts []*model.Endpoint, err error) {
```

Comment:
Lets be more explicit in the method declaration, keep the existing method `getEndpoints` and add `getCurrentClusterEndpoints` for readability.